<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AIPrevisionWidget</class>
 <widget class="QWidget" name="AIPrevisionWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>AI Prevision</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="labelTitle">
     <property name="text">
      <string>AI Prevision Algorithms</string>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="tableAlgorithms">
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <column>
      <property name="text">
       <string>Algorithm</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Description</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupParameters">
     <property name="title">
      <string>Hough Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayoutParams">
      <item row="0" column="0">
       <widget class="QLabel" name="labelThreshold">
        <property name="text">
         <string>Threshold:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="spinThreshold">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>500</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelMinLength">
        <property name="text">
         <string>Min Length:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinMinLength">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>500</number>
        </property>
        <property name="value">
         <number>30</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelMaxGap">
        <property name="text">
         <string>Max Gap:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="spinMaxGap">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelAngleTolerance">
        <property name="text">
         <string>Angle Tolerance:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="spinAngleTolerance">
        <property name="minimum">
         <double>1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>45.000000000000000</double>
        </property>
        <property name="value">
         <double>30.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelClusterDist">
        <property name="text">
         <string>Cluster Distance:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="spinClusterDist">
        <property name="minimum">
         <double>1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutButtons">
     <item>
      <widget class="QPushButton" name="btnApply">
       <property name="text">
        <string>Apply (Edge)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnHoughLines">
       <property name="text">
        <string>Hough Lines</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnClustering">
       <property name="text">
        <string>Clustering</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnFindIntersection">
       <property name="text">
        <string>Find Intersection</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnReset">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutRunAll">
     <item>
      <widget class="QPushButton" name="btnRunAll">
       <property name="text">
        <string>Run All (Edge + Hough + Clustering + Intersection)</string>
       </property>
       <property name="styleSheet">
        <string>background-color: #4CAF50; color: white; font-weight: bold;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnShowOriginalSize">
       <property name="text">
        <string>Show Original Size</string>
       </property>
       <property name="styleSheet">
        <string>background-color: #2196F3; color: white; font-weight: bold;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutSaveLoad">
     <item>
      <widget class="QPushButton" name="btnSaveParams">
       <property name="text">
        <string>Save Parameters</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnLoadParams">
       <property name="text">
        <string>Load Parameters</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupResult">
     <property name="title">
      <string>Result</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayoutResult">
      <item>
       <widget class="QLabel" name="labelResult">
        <property name="text">
         <string>Intersection: X=0, Y=0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelOriginalCoord">
        <property name="text">
         <string>Original: X=0, Y=0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelExecutionTime">
        <property name="text">
         <string>Execution Time: 0 ms</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelServerTime">
        <property name="text">
         <string>Server Time: 0 ms</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
